docs/developer/clients.rst

```diff
@@ -2,27 +2,321 @@
 
 .. _clients:
 
-Clients
-=======
-Multi-Agent interactions are orchestrated by OCS Clients. Clients can be
+Clients and Control Programs
+============================
+
+The OCS Client ("Client") is used to command and control a corresponding OCS
+Agent ("Agent"). Conceptually, this is the piece of OCS that connects to the
+crossbar server and makes the "remote procedure call" to an Agent. Multiple
+Clients can be used to orchestrate an observatory within a Control Program
+("program" or "script").
+
+While there are other options for how a Client could be implemented, the focus
+of this documentation will be on using the OCSClient object within a script.
+Other options are discussed below.
+
+.. _ocs_client:
+
```

Comment:
This (somewhere in Clients or OCSClient) could/should mention ocs-client-cli.  And ocs-client-cli docs could link back here.